packages/gateway/src/workflow/index.ts

```diff
@@ -129,13 +135,32 @@ export async function createRegistration(
   event: EVENT_TYPE,
   authHeader: IAuthHeader
 ) {
-  const res = await createRequest<{
-    compositionId: string
-    trackingId: string
-    isPotentiallyDuplicate: boolean
-  }>('POST', '/create-record', authHeader, { record, event })
+  if (inScope(authHeader, [USER_SCOPE.REGISTER])) {
```

Comment:
Remember how the "create record" handler is basically 3 operations in one? I split it and lifted the detection to this level. Ideally the client would call right graphql handler from the start